cms/djangoapps/contentstore/api/views/utils.py

```diff
@@ -135,3 +138,19 @@ def _wrapper_view(self, request, course_id, *args, **kwargs):
             )
         return view(self, request, course_key, *args, **kwargs)
     return _wrapper_view
+
+
+class NumericalInputValidationView(GenericAPIView):
+    """Class in charge of NumericalInputValidations"""
+    def post(self, request):
+        """function to validate a math expression (formula) and return of the numeric input is valid or not"""
+        result = {'preview': '',
+                  'is_valid': True,
+                  'error': ''}
+        try:
+            result['preview'] = latex_preview(request.data.get('formula'))
+        except pyparsing.ParseException:
+            result["error"] = "Sorry, couldn't parse formula"
+            result['is_valid'] = False
+            return Response(result, status=400)
```

Comment:
I think that it would be a better API contract if we returned 400 only if the request itself was bad (for example, if it was missing the `formula` field, or the formula was an array instead of a string, etc.)

In the case that the request is proper but the formula is bad, I recommend returning 200 but with `is_valid: False` in the response. That way, frontends which call this API can always just check the value of `is_valid` instead of having to check the return code _and_ the valid of `is_valid`.